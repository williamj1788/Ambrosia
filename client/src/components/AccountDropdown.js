import React from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import { clearUser } from '../redux/action'
import s from '../styles/AccountDropdown.module.scss';

class AccountDropdown extends React.Component{
    
    state = {
        redirect: false,
    }
    
    signOutUser = () => {
        fetch('/api/user/signout', {
            credentials: 'include',
        }).then(() => {
            this.props.dispatch(clearUser());
        });
    }

    redirectToOrder = () => {
        this.setState({
            redirect: true,
        });
    }

    componentDidUpdate(){
        if(this.state.redirect){
            this.setState({
                redirect: false
            });
        }
    }
    
    render(){
        if(this.state.redirect){
            return <Redirect to='/user/orders' />
        }
        
        
        let style = {
            height: window.innerWidth > 1000 ? '100px' : '50px'
        };
        if(!this.props.show){
            if(window.innerWidth > 1000){
                style.height = '0';
            }else{
                style.width = '0';
            }
        }

        return(
            <div className={s.dropDown} style={style}>
                <div className={s.content}>
                    <div className={s.tab} onClick={this.redirectToOrder} >
                        <span>Order History</span>
                    </div>
                    <div className={s.tab} onClick={this.signOutUser} >
                        <span>Sign Out</span>
                    </div>
                </div>
            </div>
        )
    }
}

export default connect(state => {
    return {user: state.user}
})(AccountDropdown);