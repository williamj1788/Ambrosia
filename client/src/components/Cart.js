import React from 'react';
import s from '../styles/Cart.module.scss';
import { connect } from 'react-redux';
import { editOrder, removeOrder } from '../redux/action';
import { FaTimes } from 'react-icons/fa';
import uuid from 'uuid';
import { Redirect, withRouter } from 'react-router-dom';

class Cart extends React.Component{
    
    state = {
        active: 1,
        redirect: false,
        error: null,
    }

    handleClick = () => {
        if(!this.props.user){
            this.setRedirect('/signup');
        }else if(this.props.orders.length <= 0){
            this.setState({
                error: 'There are no items'
            });
        }else{
            this.setRedirect('/order');
        }
    }

    removeOrder = id => {
        this.props.dispatch(removeOrder(id));
    }

    editOrder = (id, qty) => {
        this.props.dispatch(editOrder({id,qty}));
    }


    setActive = value => {
        this.setState({active: value});
    }

    setRedirect = value => {
        this.setState({redirect: value});
    }
    
    render(){
        const { redirect, active, error } = this.state;
        const { orders, location, toggle } = this.props;
        if(redirect){
            if(redirect === '/signup' && location.pathname !== '/signup'){
                return <Redirect push to={redirect} />
            }else if(redirect === '/order' && location.pathname !== '/order'){
                return <Redirect push to={redirect} />
            }else{
                toggle();
            }
        }

        return(
            <div className={s.dark}>
                <div className={s.cart}>
                    <Header toggle={toggle} />
                    <OrderContainer 
                    orders={this.props.orders} 
                    remove={this.removeOrder}
                    edit={this.editOrder}
                    active={active}
                    setActive={this.setActive}  />
                    <Footer orders={orders} onClick={this.handleClick} error={error} />
                </div>
            </div>
        )
    }
}

const Header = ({ toggle }) => {
    return(
        <div className={s.header} >
            <span>Cart</span>
            <div className={s.close} onClick={toggle} >
                <FaTimes size='2em' />
            </div>
        </div>
    )
}

const Footer = ({ orders, onClick, error }) => {
    let subTotal = orders.reduce((acc, val) => {
        return acc + (val.price * val.qty);
    }, 0);

    let tax = subTotal * 0.06;

    let deliveryFee = 3;

    let total = subTotal + tax + deliveryFee;

    return(
        <div className={s.footer}>
            <div className={s.flexTotal}>
                <span>Sub Total:</span>
                <span>{'$' + subTotal.toFixed(2)}</span>
            </div>
            <div className={s.flexTotal}>
                <span>Tax(6%):</span>
                <span>{'$' + tax.toFixed(2)}</span>
            </div>
            <div className={s.flexTotal}>
                <span>Delivery fees:</span>
                <span>{'$' + deliveryFee.toFixed(2)}</span>
            </div>
            <div className={s.flexTotal}>
                <span>Total:</span>
                <span>{'$' + total.toFixed(2)}</span>
            </div>
            <button onClick={onClick} type='button' className={s.button} >Checkout</button>
            {error && <p className={s.error}>{error}</p>}
        </div>
    )
}

const OrderContainer = ({ orders, remove, edit, active, setActive }) => {
    const MaxOrders = 7;
    let  curOrders = orders.slice((MaxOrders * (active - 1)), (MaxOrders * active));
    curOrders = curOrders.map((order) => {
        return <Order key={order.id} remove={remove} edit={edit} {...order} />
    });

    if(curOrders.length < 1 && active !== 1){
        setActive(active - 1);
    }

    let counters = [];
    for(let i = 1; i <= Math.ceil(orders.length / MaxOrders); i++){
        counters.push(<button key={uuid()} onClick={() => setActive(i)} className={`${s.counter} ${active === i && s.active}`}>{i}</button>);
    };
    
    return(
        <div className={s.OrderContainer}>
            {curOrders}
            <div className={s.counterCounter}>
                {counters.length > 1 && counters}
            </div>
        </div>
    )
};

const Order = ({ name, price, qty, id, remove, edit }) => {
    return(
        <div className={s.order}>
            <div className={s.orderInfo}>
                <p className={s.orderTitle}>{name}</p>
                <p className={s.orderPrice}>{'$' + price.toFixed(2)}</p>
                <select className={s.select} onChange={event => edit(id, event.target.value)} defaultValue={qty} name="Qty">
                    <option value="1">1</option>
                    <option value="2">2</option>
                    <option value="3">3</option>
                    <option value="4">4</option>
                    <option value="5">5</option>
                    <option value="6">6</option>
                    <option value="7">7</option>
                </select>
            </div>
            <button type='button' className={s.orderRemove} onClick={() => remove(id)} >Remove</button>
        </div>
    )
}

export default withRouter(connect(state => {
    return{
        user: state.user,
        orders: state.orders
    }
})(Cart));