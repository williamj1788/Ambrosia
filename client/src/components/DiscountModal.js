import React from 'react';
import s from '../styles/Products.module.scss';
import { FaTimes } from "react-icons/fa";

class DiscountModal extends React.Component{
    render(){
        return(
            <div className={s.dark}>
                <div className={s.discountModal}>
                    <div className={s.header} style={{backgroundColor: '#0033ff'}} >
                        <span style={{color: 'white'}} >Add A Discount</span>
                        <div className={s.close} onClick={this.props.show}>
                            <FaTimes size="1.75em" />
                        </div>
                    </div>
                    <p className={s.discountTitle}>Product - Pizza</p>
                    <div className={s.discountFormRecord}>
                        <p className={s.discountLabel}>Old Price:</p>
                        <p className={s.discountRignt}>12.99</p>
                    </div>
                    <form >
                        <div className={s.discountFormRecord}>
                            <label className={s.discountLabel} htmlFor="newPrice">New Price:</label>
                            <input className={s.discountRignt} type="number" name="newPrice" />
                        </div>
                        <div className={s.discountFormRecord}>
                            <label className={s.discountLabel} htmlFor="newPrice">Expire in:</label>
                            <select className={s.discountRignt} name="expireIn">
                                <option value="1">1 day</option>
                                <option value="2">2 days</option>
                                <option value="3">3 days</option>
                                <option value="4">4 days</option>
                                <option value="5">5 days</option>
                                <option value="6">6 days</option>
                                <option value="7">7 days</option>
                            </select>
                        </div>
                        <button className={s.discountButton} type='submit'>Add</button>
                    </form>
                    
                </div>
            </div>
        )
    }
}

export default DiscountModal;