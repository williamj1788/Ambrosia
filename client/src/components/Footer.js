import React from 'react';
import s from '../styles/Footer.module.scss';

import twitterIcon from '../images/twitter_icon.png';
import facebookIcon from '../images/Facebook-Icon.png';
import instagramIcon from '../images/instagram.png';

import locationIcon from '../images/location-icon.png';
import phoneIcon from '../images/phone-icon.png';
import emailIcon from '../images/email-icon.svg';

const Footer = () => {
    return(
        <footer className={s.footer}>
            <div className={s.container}>
                <div className={s.opening}>
                    <h4 className={s.title}>Opening Hours</h4>
                    <div className={s.hours}>
                        <div className={s.hoursSection}>
                            <p className={s.hoursTitle}>Monday - Friday</p>
                            <p className={s.hoursText}>8.00 am - 10.00 pm</p>
                        </div>
                        <div className={s.hoursSection}>
                            <p className={s.hoursTitle}>Saturday</p>
                            <p className={s.hoursText}>10.00 am - 5.00 pm</p>
                        </div>
                        <div className={s.hoursSection}>
                            <p className={s.hoursTitle}>Sunday</p>
                            <p className={s.hoursText}>Closed</p>
                        </div>
                    </div>
                </div>
                <div className={s.contact}>
                    <h4 className={s.title}>Contact Us</h4>
                    <div className={s.contactSocialIconContainer}>
                        <img className={s.contactSocialIcon} src={twitterIcon} alt="Social Media Icon"/>
                        <img className={s.contactSocialIcon} src={facebookIcon} alt="Social Media Icon"/>
                        <img className={s.contactSocialIcon} src={instagramIcon} alt="Social Media Icon"/>
                    </div>
                    <div className={s.contactSection}>
                        <img className={s.contactSectionIcon} src={locationIcon} alt="Location Icon"/>
                        <p className={s.contactSectionText}>203 Fake St. Mountain View, San Francisco, California, USA</p>
                    </div>
                    <div className={s.contactSection}>
                        <img className={s.contactSectionIcon} src={phoneIcon} alt="Phone Icon"/>
                        <p className={s.contactSectionText}>+1 978-236-7889</p>
                    </div>
                    <div className={s.contactSection}>
                        <img className={s.contactSectionIcon} src={emailIcon} alt="Email Icon"/>
                        <p className={s.contactSectionText}>JacquezWilliams@teleworm.us</p>
                    </div>
                </div>
            </div>
        </footer> 
    )
}

export default Footer;