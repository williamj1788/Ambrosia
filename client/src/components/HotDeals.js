import React from 'react';
import s from '../styles/HotDeals.module.scss';
import PizzaIcon from '../images/Pizza_icon_white.png';
import { connect } from 'react-redux';

export class HotDeals extends React.Component{

    findProductsWithDiscounts = () => {
        return this.props.products.filter(product => {
            return product.discountObj.length !== 0;
        })
    }
    
    render(){
        return(
            <section className={s.hotDeals}>
                <h3 className={s.hotDealsTitle}>Hot Deals</h3>
                <DealContainer products={this.findProductsWithDiscounts()} />
            </section>
        )
    }
}

export const DealContainer = ({ products }) => {
    const deals = products.map(product => {
        return <Deal {...product} />
    });


    return(
        <div className={s.dealsContainer}>
            {deals}
        </div>
    )
}


export const Deal = ({ name, picture, description, price, discountObj }) => {
    return(
        <div className={s.deal}>
            <img className={s.dealImg} src={picture} alt="product"/>
            <div className={s.dealInfo}>
                <span className={s.dealTitle}>{name}</span>
                <p className={s.dealDesc}>{description}</p>
                <div className={s.dealStat}>
                    <span className={s.dealPrice}>{getDiscountPercent(discountObj[0].price, price) + '% off'}</span>
                    <span className={s.dealPrice}>{Math.floor(1 + daysBetweenDates(discountObj[0].expiresAt, new Date())) + ' days left'}</span>
                </div>
                <div className={s.dealPriceContainer}>
                    <span className={s.dealPrice}><s>{'$' + price}</s></span>
                    <span className={s.dealPrice}>{'$' + discountObj[0].price}</span>
                    <button className={s.dealButton}>Place Order</button>
                </div>
            </div>
        </div>
    )
}

function roundNumber(num, target = 2){
    return Math.round(num * Math.pow(10, target)) / 100
};

function getDiscountPercent(discount, price) {
   return Math.round((1 - roundNumber(discount / price)) * 100)
}

function daysBetweenDates(day1, day2) {
    console.log(day1, day2)
    console.log(Math.round(((new Date(day1).getTime() - day2.getTime()) / (1000 * 60 * 60 * 24)) * 100) / 100)
    return Math.round(((new Date(day1).getTime() - day2.getTime()) / (1000 * 60 * 60 * 24)) * 100) / 100
}

export default connect(state => {
    return {
        products: state.products
    }
})(HotDeals);