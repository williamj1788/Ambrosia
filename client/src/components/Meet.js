import React from 'react';
import Navbar from './shared/Navbar';
import Content from './shared/Content';
import Title from './shared/Title';

import s from '../styles/Meet.module.scss';
import chef from '../images/chef.jpg';

export class Meet extends React.Component{
    render(){
        return(
            <div>
                <Navbar />
                <Content>
                    <Title style={{margin: '50px 0 0 0'}}>Meet The Chefs</Title>
                    <div className={s.chiefContainer}>
                        <Chef 
                        img={chef}
                        name='joe down'
                        desc='Lorem ipsum dolor sit amet, 
                        consectetur adipiscing elit. Nullam sit amet purus porta, 
                        dapibus felis in malesuada diam. Integer quis imperdiet 
                       s arcu, ut ornare sem. Phasellus magna justo, tincidunt quis'
                        />
                        <Chef 
                        img={chef}
                        name='joe down'
                        desc='Lorem ipsum dolor sit amet, 
                        consectetur adipiscing elit. Nullam sit amet purus porta, 
                        dapibus felis in malesuada diam. Integer quis imperdiet 
                        arcu, ut ornare sem. Phasellus magna justo, tincidunt quis'
                        />
                        <Chef 
                        img={chef}
                        name='joe down'
                        desc='Lorem ipsum dolor sit amet, 
                        consectetur adipiscing elit. Nullam sit amet purus porta, 
                        dapibus felis in malesuada diam. Integer quis imperdiet 
                        arcu, ut ornare sem. Phasellus magna justo, tincidunt quis'
                        />
                        <Chef 
                        img={chef}
                        name='joe down'
                        desc='Lorem ipsum dolor sit amet, 
                        consectetur adipiscing elit. Nullam sit amet purus porta, 
                        dapibus felis in malesuada diam. Integer quis imperdiet 
                        arcu, ut ornare sem. Phasellus magna justo, tincidunt quis'
                        />
                    </div>
                </Content>
            </div>
        )
    }
}

export const Chef = ({ img, name, desc }) => {
    return(
        <div className={s.chief}>
            <img className={s.chiefImg} src={img} alt={name}/>
            <p className={s.chiefDesc}>{desc}</p>
        </div>
    )
}

export default Meet;