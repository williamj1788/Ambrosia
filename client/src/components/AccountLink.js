import React from 'react';
import { connect } from 'react-redux';
import userIcon from '../images/UserIcon.png';
import UserArrow from '../images/Account-arrow.png';
import s from '../styles/Navbar.module.scss';
import { NavLinkWrapper } from './NavComponents';
import AccountDropdown from './AccountDropdown';

export class AccountLink extends React.Component{
    
    state = {
        showDropdown: false,
    }

    componentDidMount(){
        window.addEventListener('scroll', this.handleScroll);
    }

    handleScroll = () => {
        if(this.state.showDropdown){
            this.setState({
                showDropdown: false,
            })
        }
    }
    
    toggleDropdown = () =>{
        this.setState({
            showDropdown: !this.state.showDropdown
        });
    }

    render(){
        const { user } = this.props;
        const { showDropdown } = this.state
        if(user){
            return(
                <div style={{position: 'relative'}} >
                    <div className={s.navLink} style={{cursor: 'pointer'}} onClick={this.toggleDropdown} >
                        <img className={s.linkLogo} src={user.picture ? user.picture : userIcon} alt="User Icon" />
                        <span>{user.firstname}</span>
                        <img className={s.userArrow} src={UserArrow} alt="User Arrow"/>
                    </div>
                    <AccountDropdown show={showDropdown} />
                </div>
            )
        }
        return(
            <NavLinkWrapper to='/login' icon={userIcon}>Login</NavLinkWrapper>
        )
    }
}



export default connect(state => {
    return {user: state.user}
})(AccountLink);