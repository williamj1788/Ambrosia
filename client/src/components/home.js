import React from 'react';

import Navbar from './Navbar';
import ReviewSection from './ReviewSection';
import Footer from './Footer';
import HotDeals from './HotDeals';

import s from '../styles/Home.module.scss';

import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';

import PizzaIcon from '../images/Pizza_icon_white.png';
import MoneyIcon from '../images/Money_icon.png';
import TruckIcon from '../images/truck_icon.png';
import ArrowDown from '../images/arrow-down.png';


export class Home extends React.Component{
    
    state = {
        redirectToMenu: false
    }

    setRedirect = () => {
        this.setState({
            redirectToMenu: true,
        });
    }
    
    render(){
        if(this.state.redirectToMenu){ 
            return <Redirect to='/menu' />
        }
        return(
            <div>
                <Hero redirect={this.setRedirect}/>
                <About />
                <HotDeals />
                <ReviewSection />
                <Footer />
                <Copyright />
            </div>
        )
    }
}

const Hero = ({ redirect }) => {
    return(
        <section className={s.hero}>
            <div className={s.darken}>
                <Navbar />
                <p className={s.heroTitle}>Best Pizza in Town</p>
                <button onClick={redirect} className={s.heroButton}>View Menu</button>
                <img className={s.heroArrow} src={ArrowDown} alt="ArrowDown Icon"/>
            </div>
        </section>
    )
}

const About = () => {
    return(
        <section className={s.about}>
            <AboutTitle />
            <div className={s.flexContainer} >
                <Unit 
                img={PizzaIcon}
                title={'Great Pizza'}
                desc={'We go through multiple checks of each ingredient to ensure that our pizza is top quality'}
                />
                <Unit 
                img={MoneyIcon}
                title={'Low Prices'}
                desc={'We work with our local farmers to provide affordable and quality ingredients and pass the savings to our customers'}
                />
                <Unit 
                img={TruckIcon}
                title={'Fast Service'}
                desc={'With active drivers that are always on standby, we guarantee a delivery in under 30 mins or the order is free'}
                />
            </div>
        </section>
    )
}

const AboutTitle = () => {
    return(
        <div>
            <h2 className={s.aboutTitle}>About Us</h2>
            <p className={s.aboutDesc}>We are committed to providing our customers with excellent service that makes us deserving of being the best pizzeria in town</p> 
        </div>
    )
}

export const Unit = ({ img, title, desc }) => {
    return(
        <div className={s.AboutUnits}>
            <img className={s.aboutUnitImg} src={img} alt="Icon"/>
            <span className={s.aboutUnitTitle} >{title}</span>
            <p className={s.aboutUnitDesc}>{desc}</p>
        </div>
    )
}

const Copyright = () => {
    return(
        <div className={s.copyright}>
            <p className={s.copyrightText}>Copyright Â©2019 All rights reserved</p>
        </div>
    )
}

export default connect()(Home);