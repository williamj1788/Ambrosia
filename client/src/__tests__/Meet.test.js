import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Meet, Chief } from "../components/Meet";

describe('<Meet />', () => {
    test('Snapshot Test', () => {
        const component = shallow(<Meet />);
        expect(toJson(component)).toMatchSnapshot();
    });
});


describe('<Chief />', () => {
    test('Snapshot Test', () => {
        const component = shallow(<Chief img={'mock img'} name={'mock name'} desc={'mock desc'} />);
        expect(toJson(component)).toMatchSnapshot();
    });

    test('should render img and p element with correct props', () => {
        const component = shallow(<Chief img={'mock img'} name={'mock name'} desc={'mock desc'} />);
        expect(component.find('img').props().src).toEqual('mock img');
        expect(component.find('img').props().alt).toEqual('mock name');
        expect(component.find('p').props().children).toEqual('mock desc');
    });
});