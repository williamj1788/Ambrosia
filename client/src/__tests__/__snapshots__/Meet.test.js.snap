// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Chief /> Snapshot Test 1`] = `
<div
  className="chief"
>
  <img
    alt="mock name"
    className="chiefImg"
    src="mock img"
  />
  <p
    className="chiefDesc"
  >
    mock desc
  </p>
</div>
`;

exports[`<Meet /> Snapshot Test 1`] = `
<div>
  <ConnectFunction />
  <div
    className="content"
  >
    <h1
      className="title"
    >
      Meet The Cheifs
    </h1>
    <div
      className="chiefContainer"
    >
      <Chief
        desc="Lorem ipsum dolor sit amet, 
 consectetur adipiscing elit. Nullam sit amet purus porta, 
 dapibus felis in malesuada diam. Integer quis imperdiet 
 arcu, ut ornare sem. Phasellus magna justo, tincidunt quis"
        img="chef.jpg"
        name="joe down"
      />
      <Chief
        desc="Lorem ipsum dolor sit amet, 
 consectetur adipiscing elit. Nullam sit amet purus porta, 
 dapibus felis in malesuada diam. Integer quis imperdiet 
 arcu, ut ornare sem. Phasellus magna justo, tincidunt quis"
        img="chef.jpg"
        name="joe down"
      />
      <Chief
        desc="Lorem ipsum dolor sit amet, 
 consectetur adipiscing elit. Nullam sit amet purus porta, 
 dapibus felis in malesuada diam. Integer quis imperdiet 
 arcu, ut ornare sem. Phasellus magna justo, tincidunt quis"
        img="chef.jpg"
        name="joe down"
      />
      <Chief
        desc="Lorem ipsum dolor sit amet, 
 consectetur adipiscing elit. Nullam sit amet purus porta, 
 dapibus felis in malesuada diam. Integer quis imperdiet 
 arcu, ut ornare sem. Phasellus magna justo, tincidunt quis"
        img="chef.jpg"
        name="joe down"
      />
    </div>
  </div>
</div>
`;
